providers/azure.go

```diff
@@ -209,6 +210,8 @@ func (p *AzureProvider) EnrichSession(ctx context.Context, session *sessions.Ses
 		session.Email = email
 	}
 
+	p.setUser(ctx, session)
```

Comment:
proper error handling will be necessary here. as the username wasn't enforced before. just logging it but otherwise continuing the enrichment of the session would be fine for me.